### PR TITLE
Upgrade to release 0.1.21

### PR DESCRIPTION
**Pull request by dallmeyer**
One issue I initially ran into while running `extractor.exe` was with [this assertion](https://github.com/open-goal/jak-project/blob/7d5901aa95836732190feaf06cd3f4477fb73340/common/type_system/TypeSystem.cpp#L207-L211), it seems because we added `reset!` to `game-info` [here](https://github.com/Zedb0T/OpenGoal-CheckpointRandomizer/commit/c0fb79cbf57fba40b18520918edb0d4b5d98c820#diff-246dc7f3d232821668c2570335fa23dc9131c592a9b23d234c12817dc36b2a2a)?

I just [commented out that assertion](https://github.com/open-goal/jak-project/compare/master...dallmeyer:jak-project:rando?expand=1#diff-e4f9d40bc0b1ce906abde6b490e99fefaebb55ca22819311f27ab60bad6787bf) and was able to get everything extracting/decompiling/building after that. Not sure if you did this as well when you built your exe's but I wasn't able to reproduce this issue with the current version of `extractor.exe` in main.

Should be mostly clean merge with 0.1.21 and some minor cleanup. Also includes the change from #65. Oh and probably want to [hide whitespace changes](https://github.com/Zedb0T/OpenGoal-CheckpointRandomizer/pull/66/files?w=1)

### CODE REVIEW

File: Checkpoint randomizer/data/goal_src/jak1/engine/game/game-info.gc [hunk 1]
```diff
@@ -775,21 +775,21 @@
...
+          (begin
+            ;;orbless mod
+            (if (-> *challenges-settings* challenges-orbless?)
+            (begin 
+              (reset-all)
+              (return 1.0)
+            )
+            (begin
+             (sound-play "money-pickup")
+            )
```
Comment (dallmeyer) [inline Checkpoint randomizer/data/goal_src/jak1/engine/game/game-info.gc:787]: can revert this too if you want, wasn't sure if it was this getting cut off or the eels sound that I was hearing lol

File: Checkpoint randomizer/data/goal_src/jak1/engine/game/main.gc [hunk 2]
```diff
@@ -17,10 +17,10 @@
 (none))
 
 (defun reset-all ()
-(dotimes (i 205) (sound-play-by-name (static-sound-name "caught-eel") (new-sound-id) 1024 0 0 (sound-group sfx) #t))
+;;(dotimes (i 205) (sound-play-by-name (static-sound-name "caught-eel") (new-sound-id) 1024 0 0 (sound-group sfx) #t))
```
Comment (dallmeyer) [inline Checkpoint randomizer/data/goal_src/jak1/engine/game/main.gc:20]: 🎣

Comment (Zedb0T) [reply Checkpoint randomizer/data/goal_src/jak1/engine/game/main.gc:20]: I love this part :( can we do a if dallmeyer variable haha
I think it enhances the "oh no" effect of collecting an orb in orbless

File: Checkpoint randomizer/data/goal_src/jak1/pc/progress-pc.gc [hunk 3]
```diff
@@ -134,18 +152,18 @@
...
 
 (define *misc-options* (new 'static 'boxed-array :type game-option
+    (new 'static 'game-option :option-type (game-option-type on-off) :name (game-text-id money-starburst) :scale #t)
+    (new 'static 'game-option :option-type (game-option-type on-off) :name (game-text-id discord-rpc) :scale #t)
     (new 'static 'game-option :option-type (game-option-type menu) :name (game-text-id checkpoint-randomizer) :scale #t :param3 (game-option-menu randomizer-main-options))
     (new 'static 'game-option :option-type (game-option-type menu) :name (game-text-id custom-challenges) :scale #t :param3 (game-option-menu challenges-options))
     (new 'static 'game-option :option-type (game-option-type menu) :name (game-text-id mod-options) :scale #t :param3 (game-option-menu mod-options))
-    (new 'static 'game-option :option-type (game-option-type menu) :name (game-text-id disable-auto-save) :scale #t :param3 (game-option-menu memcard-disable-auto-save))
-    (new 'static 'game-option :option-type (game-option-type on-off) :name (game-text-id discord-rpc) :scale #t)
     (new 'static 'game-option :option-type (game-option-type button) :name (game-text-id back) :scale #t)
```
Comment (Zedb0T) [inline Checkpoint randomizer/data/goal_src/jak1/pc/progress-pc.gc:167]: Did you do this? or is this an effect of a merge or something??

**Review by Zedb0T — APPROVED** [review]

**Pull request merged**